spec/unit/webrtc/groupCall.spec.ts

```diff
@@ -14,85 +14,305 @@ See the License for the specific language governing permissions and
+                        // since we cheat & use the same maps for each, we can
+                        // just add it once.
+                        client2Room.currentState.events.set(eventType, subMap);
+                    }
+                    subMap.set(statekey, fakeEvent);
+
+                    groupCall1.onMemberStateChanged(fakeEvent);
+                    groupCall2.onMemberStateChanged(fakeEvent);
+                }
+                return Promise.resolve(null);
+            };
+
+            const client1Room = new Room(FAKE_ROOM_ID, client1, FAKE_USER_ID_1);
+
+            const client2Room = new Room(FAKE_ROOM_ID, client2, FAKE_USER_ID_2);
+
+            groupCall1 = new GroupCall(
+                client1, client1Room, GroupCallType.Video, false, GroupCallIntent.Prompt, FAKE_CONF_ID,
+            );
+
+            groupCall2 = new GroupCall(
+                client2, client2Room, GroupCallType.Video, false, GroupCallIntent.Prompt, FAKE_CONF_ID,
+            );
+
+            client1Room.currentState.members[FAKE_USER_ID_1] = {
+                userId: FAKE_USER_ID_1,
+            } as unknown as RoomMember;
+            client1Room.currentState.members[FAKE_USER_ID_2] = {
+                userId: FAKE_USER_ID_2,
+            } as unknown as RoomMember;
+
+            client2Room.currentState.members[FAKE_USER_ID_1] = {
+                userId: FAKE_USER_ID_1,
+            } as unknown as RoomMember;
+            client2Room.currentState.members[FAKE_USER_ID_2] = {
+                userId: FAKE_USER_ID_2,
+            } as unknown as RoomMember;
+        });
+
+        afterEach(function() {
+            MockRTCPeerConnection.resetInstances();
+        });
+
+        it("Places a call to a peer", async function() {
+            await groupCall1.create();
+
+            try {
+                const mockSendToDevice = jest.fn();
+
+                const toDeviceProm = new Promise(resolve => {
+                    mockSendToDevice.mockImplementation(resolve);
+                });
+
+                client1.sendToDevice = mockSendToDevice;
+
+                await groupCall1.enter();
+
+                await groupCall2.enter();
+
+                MockRTCPeerConnection.triggerAllNegotiations();
+
+                await toDeviceProm;
+
+                expect(mockSendToDevice.mock.calls[0][0]).toBe("m.call.invite");
+
+                const toDeviceCallContent = mockSendToDevice.mock.calls[0][1];
+                expect(Object.keys(toDeviceCallContent).length).toBe(1);
+                expect(Object.keys(toDeviceCallContent)[0]).toBe(FAKE_USER_ID_2);
+
+                const toDeviceBobDevices = toDeviceCallContent[FAKE_USER_ID_2];
+                expect(Object.keys(toDeviceBobDevices).length).toBe(1);
+                expect(Object.keys(toDeviceBobDevices)[0]).toBe(FAKE_DEVICE_ID_2);
+
+                const bobDeviceMessage = toDeviceBobDevices[FAKE_DEVICE_ID_2];
+                expect(bobDeviceMessage.conf_id).toBe(FAKE_CONF_ID);
```

Comment:
If possible, this is another set of assertions that might be more readable as a single `expect(mockSendToDevice).toHaveBeenLastCalledWith(...)`